step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/automation/YamlFunctionGroup.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (C) 2020, exense GmbH
+ *
+ * This file is part of STEP
+ *
+ * STEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * STEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with STEP.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package step.artefacts.automation;
+
+import step.artefacts.FunctionGroup;
+import step.core.dynamicbeans.DynamicValue;
+
+public class YamlFunctionGroup extends YamlTokenSelector<FunctionGroup> {
+
+    private DynamicValue<String> dockerImage = new DynamicValue<>();
+
+    private DynamicValue<String> containerUser = new DynamicValue<>();
+
+    private DynamicValue<String> containerCommand = new DynamicValue<>();
+
+    public YamlFunctionGroup() {
+        super(FunctionGroup.class);
+    }
+
+    @Override
+    protected void fillArtefactFields(FunctionGroup res) {
+        super.fillArtefactFields(res);
+        if (dockerImage != null) {
```

Comment:
Can't we do these not null checks in a generic manner?

Reply:
@jeromecomte 
I'll try to write some method to check/copy all fields in model by reflection. Then all this checks and copying will be perfomed implicitly.

Reply:
@jeromecomte @david-stephan 

I've added the `AbstractYamlModel` class with `copyFieldsToObject` and `copyFieldsFromObject` methods to copy all fields automatically. If the custom logic is required for some field, this field should be marked with new `@YamlFieldCustomCopy`. 
The new solution is applied in all artefact models now.

Reply:
@iegorov777 looks good